src/framework/components/rigid-body/system.js

```diff
@@ -552,6 +560,222 @@ class RigidBodyComponentSystem extends ComponentSystem {
+     * @param {Vec3} [rotation] - The world space rotation for the cylinder to have.
+     *
+     * @returns {RaycastResult[]} An array of cylindercast hit results (0 length if there were no hits).
+     */
+    cylinderCast(radius, height, axis, position, rotation) {
+        let fn = 'btCylinderShape';
+
+        if (axis === 0) {
+            fn = 'btCylinderShapeX';
+        } else if (axis === 2) {
+            fn = 'btCylinderShapeZ';
+        }
+
+        return this._shapecast(new Ammo[fn](radius, height), position, rotation);
+    }
+
+    /**
+     * Perform a collision check on the world and return all entities the sphere hits.
+     * It returns an array of {@link RaycastResult}, one for each hit. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {number} radius - The radius of the sphere.
+     * @param {Vec3} [position] - The world space position for the sphere to be.
+     * @param {Vec3} [rotation] - The world space rotation for the sphere to have.
+     *
+     * @returns {RaycastResult[]} An array of spherecast hit results (0 length if there were no hits).
+     */
+    sphereCast(radius, position, rotation) {
+        return this._shapecast(new Ammo.btSphereShape(radius), position, rotation);
+    }
+
+    /**
+     * Perform a collision check on the world and return all entities the shape hits.
+     * It returns an array of {@link RaycastResult}, one for each hit. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {Ammo.btCollisionShape} shape - The Ammo.btCollisionShape to use for collision check.
+     * @param {Vec3} [position] - The world space position for the shape to be.
+     * @param {Vec3} [rotation] - The world space rotation for the shape to have.
+     *
+     * @returns {RaycastResult[]} An array of spherecast hit results (0 length if there were no hits).
+     * @private
+     */
+    _shapecast(shape, position = Vec3.ZERO, rotation = Vec3.ZERO) {
+        Debug.assert(Ammo.ConcreteContactResultCallback, 'pc.RigidBodyComponentSystem#_shapecast: Your version of ammo.js does not expose Ammo.ConcreteContactResultCallback. Update it to latest.');
+
+        const results = [];
+
+        // Set proper position and rotation
+        ammoVec3.setValue(position.x, position.y, position.z);
+        ammoQuat.setEulerZYX(rotation.z, rotation.y, rotation.x);
+
+        // Assign position and rotation to transform.
+        ammoTransform.setIdentity();
+        ammoTransform.setOrigin(ammoVec3);
+        ammoTransform.setRotation(ammoQuat);
+
+        // We only initialize the shapecast body here so we don't have an extra body unless the user uses this function
+        if (!shapecastBody) {
+            shapecastBody = this.createBody(0, shape, ammoTransform);
+        }
+
+        // Add the body to the world.
+        this.addBody(shapecastBody, BODYGROUP_TRIGGER, BODYMASK_ALL);
```

Comment:
Check this page out:

https://andysomogyi.github.io/mechanica/bullet.html#contact-text

It says:

> The contactTest query will peform a collision test against all overlapping objects in the world, and produces the results using a callback. The query object doesn’t need to be part of the world.

Is that true?

Reply:
Ah yeah my bad, I added it as I had some issue during my testings and forgot to test without. After testing it works well with the body outside of world. Updated!